openvdb_ax/compiler/PointExecutable.h

```diff
@@ -47,72 +47,121 @@
 class PointExecutable
 {
 public:
     using Ptr = std::shared_ptr<PointExecutable>;
+    ~PointExecutable();
 
-    /// @brief Constructor
-    /// @param exeEngine Shared pointer to an llvm::ExecutionEngine object used to build functions.
-    ///        context should be the associated llvm context
-    /// @param context Shared pointer to an llvm:context object associated with exeEngine
-    /// @param attributeRegistry Registry of point attributes accessed by AX code
-    /// @param customData Custom data object which will be shared by this executable.  It can be
-    ///        used to retrieve external data from within the AX code
-    /// @param functions A map of function names to physical memory addresses which were built
-    ///        by llvm using exeEngine
-    /// @note  This object is normally be constructed by the Compiler::compile method, rather
-    ///        than directly
-    PointExecutable(const std::shared_ptr<const llvm::ExecutionEngine>& exeEngine,
-                    const std::shared_ptr<const llvm::LLVMContext>& context,
-                    const AttributeRegistry::ConstPtr& attributeRegistry,
-                    const CustomData::ConstPtr& customData,
-                    const std::unordered_map<std::string, uint64_t>& functions)
-        : mContext(context)
-        , mExecutionEngine(exeEngine)
-        , mAttributeRegistry(attributeRegistry)
-        , mCustomData(customData)
-        , mFunctionAddresses(functions) {}
+    /// @brief  Copy constructor. Shares the LLVM constructs but deep copies the
+    ///   settings. Multiple copies of an executor can be used at the same time
+    ///   safely.
+    PointExecutable(const PointExecutable& other);
 
-    ~PointExecutable() = default;
+    ////////////////////////////////////////////////////////
 
     /// @brief executes compiled AX code on target grid
-    /// @param grid Grid to apply code to
-    /// @param group Optional name of a group for filtering.  If this is not NULL,
-    ///        the code will only be applied to points in this group
-    /// @param createMissing If true, any attribute which has been used but does
-    ///        not exist will be created. Otherwise, a missing attribute runtime
-    ///        error will be thrown.
-    void execute(points::PointDataGrid& grid,
-                 const std::string* const group = nullptr,
-                 const bool createMissing = true) const;
+    void execute(points::PointDataGrid& grid) const;
+
+    ////////////////////////////////////////////////////////
+
+    /// @brief  Set a specific point group to execute over. The default is none,
+    ///   which corresponds to all points. Note that this can also be compiled
+    ///   into the AX function using the ingroup("mygroup") method.
+    /// @warning  If the group does not exist during execute, a runtime error
+    ///   will be thrown.
+    void setGroupExecution(const std::string&);
```

Comment:
should these have named args? below in setCreateMissing too

Reply:
if you are doing this the VolumeExecutable has the same

Reply:
this could be misread as the arg having no use? (tbh i didnt even realise this would compile)

Reply:
Added arg names with doxygen docs